src/main/java/seedu/modulight/model/studentscore/ScoreMatchPredicate.java

```diff
@@ -104,7 +104,7 @@ public boolean test(StudentScore score) {
         boolean isTutMatch = tutorialGroupKeywords.isEmpty() || tutorialGroupKeywords.stream()
                 .anyMatch(keyword -> score.getStudent().getTutorial().toString().equalsIgnoreCase(keyword));
         boolean isTagMatch = tagKeywords.isEmpty() || score.getTags().stream()
-                .anyMatch(tag -> tagKeywords.contains(tag.tagName.toLowerCase()));
+                .anyMatch(tag -> tagKeywords.contains(tag.tagName));
```

Comment:
May I know if there is any specific reason that you want to change the comparison to be case sensitive?